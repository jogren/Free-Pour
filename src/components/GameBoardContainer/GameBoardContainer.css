.GameBoardContainer_section {
  width: 100%
}