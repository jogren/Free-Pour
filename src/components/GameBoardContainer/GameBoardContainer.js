import React from 'react';
import GameBoardCard from '../GameBoardCard/GameBoardCard';
import './GameBoardContainer.css';

const GameBoardContainer = ({ currentDrink }) => {
  let ingredientDivs = currentDrink.ingredients.map((ingredient, index) => {
    if(ingredient.ingredient) {
      return <GameBoardCard key={index} index={index} ingredient={ingredient}/>
    }
  });

  return (
    <section>
      {ingredientDivs}
    </section>
  );
}

export default GameBoardContainer;