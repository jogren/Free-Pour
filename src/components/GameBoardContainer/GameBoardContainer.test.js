import React from 'react';
import { shallow } from 'enzyme';
import GameBoardContainer from './GameBoardContainer';

describe('GameBoardContainer', () => {
  let wrapper;
  const mockCurrentDrink = {
    name: 'Moscow Mules',
    glass: 'Copper Mug',
    instructions: 'Combine ingredients...',
    image: 'image2.png',
    ingredients: [
      {
        guessed: false,
        ingredient: 'Vodka',
        measure: '2-3 oz'
      },
      {
        guessed: false,
        ingredient: 'Ginger Ale',
        measure: '5oz'
      },
      {
        guessed: false,
        ingredient: 'Lime Juice',
        measure: '2tbsp'
      }
    ]
  }

  beforeEach(() => {
    wrapper = shallow(<GameBoardContainer 
      currentDrink={mockCurrentDrink}
    />)
  })

  it('should take a snapshot when passed the correct data', () => {
    expect(wrapper).toMatchSnapshot();
  });
})