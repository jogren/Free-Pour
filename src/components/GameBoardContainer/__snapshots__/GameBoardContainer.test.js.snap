// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoardContainer should take a snapshot when passed the correct data 1`] = `
<section>
  <GameBoardCard
    index={0}
    ingredient={
      Object {
        "guessed": false,
        "ingredient": "Vodka",
        "measure": "2-3 oz",
      }
    }
    key="0"
  />
  <GameBoardCard
    index={1}
    ingredient={
      Object {
        "guessed": false,
        "ingredient": "Ginger Ale",
        "measure": "5oz",
      }
    }
    key="1"
  />
  <GameBoardCard
    index={2}
    ingredient={
      Object {
        "guessed": false,
        "ingredient": "Lime Juice",
        "measure": "2tbsp",
      }
    }
    key="2"
  />
</section>
`;
